weather_dl/download_pipeline/parsers.py

```diff
@@ -220,7 +220,8 @@ def _number_of_replacements(s: t.Text):
 
 
 def use_date_as_directory(config: t.Dict):
-    return 'date' in config['parameters']['partition_keys'] \
+    return 'partition_keys' in config['parameters'] \
+           and 'date' in config['parameters']['partition_keys'] \
```

Comment:
Nice catch!

(It's hard to see in this diff): Is there a test for a successful config with no partition keys? If not, can you add one?